pandas/tests/arrays/string_/test_string.py

```diff
@@ -611,3 +611,11 @@ def test_setitem_scalar_with_mask_validation(dtype):
         msg = "Scalar must be NA or str"
     with pytest.raises(ValueError, match=msg):
         ser[mask] = 1
+
+
+def test_consitency_inplace():
+    expected = pd.DataFrame({"M": [""]}, dtype="string")
+    df = pd.DataFrame({"M": [""]}, dtype="string")
+    df.where(df != "", np.nan, inplace=True)
+    expected = expected.where(expected != "", np.nan)
+    tm.assert_frame_equal(expected, df)
```

Comment:
In my prior comment, I was referring to that there should be 2 `tm.assert_frame_equal` so that both `inplace=True/False` are tested separately e.g.

```
expected = pd.DataFrame({"M": [""]}, dtype="string")
df_inplace = ...
tm.assert_frame_equal(df_inplace, expected)
df_not_inplace = ...
tm.assert_frame_equal(df_not_inplace, expected)
```